index.d.ts

```diff
@@ -0,0 +1,344 @@
+ *
+ * @example
+ * interface Services {
+ *   usersPath: User
+ *   //...
+ * }
+ *
+ * declare module "vue/types/vue" {
+ *   import { FeathersVuexGlobalModels } from "feathers-vuex";
+ *   interface Vue {
+ *     $FeathersVuex: FeathersVuexGlobalModels<Services, ServicesByPath>;
+ *   }
+ * }
+ *
+ */
+interface FeathersVuexPluginType extends FeathersVuexGlobalModels { }
+
+declare module "vue/types/vue" {
+  interface Vue {
+    // $FeathersVuex: FeathersVuexPluginType;
+  }
+}
+
+/**
+ * Auth module factory
+ */
+interface FeathersVuexAuthOptions<State = FeathersVuexAuthState> extends FeathersVuexStoreOptions<State> {
+  userService: string
+}
+
+interface FeathersVuexAuthState<UserType = any> {
+  accessToken?: string, // The JWT
+  payload: Object, // The JWT payload
+
+  isAuthenticatePending: boolean,
+  isLogoutPending: boolean,
+
+  errorOnAuthenticate: Error | undefined,
+  errorOnLogout: Error | undefined,
+  user: UserType
+}
+
+interface FeathersVuexServiceState<ModelType = any, IDType = string> extends FeathersVuexCoreProps {
+  ids: IDType[],
+  keyedById: {
+    [i: string]: ModelType
+  }, // A hash map, keyed by id of each item
+  currentId?: IDType, // The id of the item marked as current
+  copy?: ModelType, // A deep copy of the current item
+  servicePath: string // The full service path
+  paginate: boolean, // Indicates if pagination is enabled on the Feathers service.
+  pagination: { [key: string]: ModelType[] }, // Indicates if pagination is enabled on the Feathers service.
+
+  isFindPending: boolean,
+  isGetPending: boolean,
+  isCreatePending: boolean,
+  isUpdatePending: boolean,
+  isPatchPending: boolean,
+  isRemovePending: boolean,
+
+  errorOnfind: Error | undefined,
+  errorOnGet: Error | undefined,
+  errorOnCreate: Error | undefined,
+  errorOnUpdate: Error | undefined,
+  errorOnPatch: Error | undefined,
+  errorOnRemove: Error | undefined
+}
+
+/**
+ * Data Components
+ */
+export const FeathersVuexFind: ComponentOptions<
+  Vue,
+  FeathersVuexFindData,
+  FeathersVuexFindMethods,
+  FeathersVuexFindComputed,
+  PropsDefinition<FeathersVuexFindProps>,
+  FeathersVuexFindProps>;
+export const FeathersVuexGet: ComponentOptions<
+  Vue,
+  FeathersVuexGetData,
+  FeathersVuexGetMethods,
+  FeathersVuexGetComputed,
+  PropsDefinition<FeathersVuexGetProps>,
+  FeathersVuexGetProps>;
+
+interface FeathersVuexFindData {
+  isFindPending: boolean;
+}
+
+interface FeathersVuexGetMethods {
+  findData(): Promise<void>;
+  fetchData(): Promise<void>;
+}
+
+interface FeathersVuexFindComputed {
+  items: any[];
+  scope: { [key: string]: any }
+  pagination: { [key: string]: any }
+}
+
+interface FeathersVuexDataComponentProps {
+  service: string;
+  query?: any;
+  queryWhen?: boolean | ((id: string, ...args: any[]) => boolean);
+  fetchQuery: any;
+  watch?: string | string[];
+  local?: boolean;
+  editScope?: ({ item: any, isGetPending: boolean }) => { [key: string]: any };
```

Comment:
At least for me this syntax is getting confused, should this perhaps be:

```typescript
 editScope?: (( item: any, isGetPending: boolean )) => { [key: string]: any };
```

Note the parenthesis instead of curly braces